src/main/java/Duke.java

```diff
@@ -1,10 +1,305 @@
+        String deadline = instruction[1];
+        Deadline newDeadline = new Deadline(task, deadline);
+        newDeadline.setType("D");
+        list.add(newDeadline);
+        printConfirmation(newDeadline, "add");
+    }
+
+    private static void eventTask(String input) throws DukeException {
+        String[] tokens = input.split("\\s+", 2);
+        String[] instruction = tokens[1].split("/");
+        if(tokens.length < 2) {
+            throw new DukeException(EMPTY_DESCRIPTION);
+        }
+        if(instruction.length < 3) {
+            throw new DukeException(WRONG_INPUTS_GIVEN);
+        }
+        String task = instruction[0];
+        String from = instruction[1];
+        String to = instruction[2];
+        String dateFrom = from.split("\\s+", 2)[1];
+        String dateTo = to.substring(to.lastIndexOf("to") + 2);
+        Event newEvent = new Event(task, dateFrom, dateTo);
+        newEvent.setType("E");
+        list.add(newEvent);
+        printConfirmation(newEvent, "add");
+    }
+
+    private static void deleteTask(String input) throws Exception {
+        String[] tokens = input.split("\\s+", 2);
+        if(tokens.length < 2) {
+            throw new DukeException(WRONG_INPUTS_GIVEN);
+        }
+        try {
+            int index = Integer.parseInt(tokens[1]) - 1;
+            Task taskToRemove = list.get(index);
+            list.remove(index);
+            printConfirmation(taskToRemove, "delete");
+        } catch (Exception e) {
+            throw new Exception(UNRECOGNISED_ITEM_INDEX);
+        }
+    }
+
+
+    private static void updateFileData() {
+        try {
+            FileWriter fw = new FileWriter(FILE_PATH);
+            for (Task t : list) {
+                String typeOfTask = t.getType();
+                String status = t.getStatus();
+                String task = t.getTask();
+                String taskDescription;
+                switch (typeOfTask) {
+                case "T":
+                    taskDescription = String.format("%s-%s-%s", typeOfTask, status, task);
+                    break;
+                case "D":
+                    Deadline deadlineTask = (Deadline) t;
+                    String deadline = deadlineTask.getDeadline();
+                    taskDescription = String.format("%s-%s-%s-%s", typeOfTask, status, task, deadline);
+                    break;
+                case "E":
+                    Event eventTask = (Event) t;
+                    String from = eventTask.getFrom();
+                    String to = eventTask.getTo();
+                    taskDescription = String.format("%s-%s-%s-from %s-to%s", typeOfTask, status, task, from, to);
+                    break;
+                default:
+                    taskDescription = "";
+                    System.out.println(FILE_UPDATING_ERROR);
+                }
+                fw.write(taskDescription + System.lineSeparator());
+            }
+            fw.close();
+        } catch (Exception e) {
```

Comment:
Nesting too deep. You can separate out writing from formatting that will solve this problem (and improve SLAP)